lib/src/shared/status.dart

```diff
@@ -120,6 +120,43 @@ class StatusCode {
   /// The request does not have valid authentication credentials for the
   /// operation.
   static const unauthenticated = 16;
+
+  /// Creates a gRPC Status code from a HTTP Status code
+  ///
+  /// Copied from: https://github.com/grpc/grpc-web/blob/master/javascript/net/grpc/web/statuscode.js
+  static int fromHttpStatus(int status) {
```

Comment:
I suggest using a map here: 

```dart
const statusMapping = {200: Status.ok, 400: StatusCode.invalidArgument, ...};

return statusMapping[status] ?? StatusCode.unknown;
```

Even better for readability would be if you do:

```dart
// It is okay to import dart:io into shared code as long as you don't use
// any runtime functionality.
import 'dart:io' show HttpStatus;

const _httpStatusToGrpcStatus = <int, int>{
  HttpStatus.ok: StatusCode.ok,
  HttpStatus.badRequest: StatusCode.invalidArgument,
  HttpStatus.unauthorized: StatusCode.unauthenticated,
  HttpStatus.forbidden: StatusCode.permissionDenied,
  HttpStatus.notFound: StatusCode.notFound,
  HttpStatus.conflict: StatusCode.aborted,
  HttpStatus.preconditionFailed: StatusCode.failedPrecondition,
  HttpStatus.tooManyRequests: StatusCode.resourceExhausted,
  HttpStatus.clientClosedRequest: StatusCode.cancelled,
  HttpStatus.internalServerError: StatusCode.unknown,
  HttpStatus.notImplemented: StatusCode.unimplemented,
  HttpStatus.serviceUnavailable: StatusCode.unavailable,
  HttpStatus.gatewayTimeout: StatusCode.deadlineExceeded,
};
```